src/auth.rs

```diff
@@ -48,12 +50,14 @@ impl Authenticator {
     }
 
     /// As the server, send a challenge to the client and validate their response.
-    pub async fn server_handshake(
+    pub async fn server_handshake<T: AsyncRead + AsyncWrite + Unpin>(
         &self,
-        stream: &mut (impl AsyncBufRead + AsyncWrite + Unpin),
+        stream: &mut Framed<T, AnyDelimiterCodec>,
```

Comment:
What do you think about making a type definition for `Delimited<T> = Framed<T, AnyDelimiterCodec>` to not repeat the `AnyDelimiterCodec` type everywhere?